#Scenes/SceneScripts/MainMenu.gd

```diff
@@ -17,13 +18,11 @@ func _ready() -> void:
 
 ## What happens when the start button is pressed
 func _on_start_pressed() -> void:
-	SaveManager.clear_data()
-	PlayerManager.clear_data()
-	MapManager.clear_map_data()
-	XpManager.clear_data()
-	CardManager.clear_data()
-	InventoryManager.reset_inventory()
-	SceneManager.goto_scene("res://#Scenes/MapUI.tscn")
+	if PlayerManager.has_saved_data():
+		start_new_game_dialog.show()
+	else:
+		MapManager.init_data()
+		SceneManager.goto_scene("res://#Scenes/MapUI.tscn")
```

Comment:
previously there was an init for SaveManager, PlayerManager, Map, XP, Card, Inventory and Scene when you start a new game. I see it's still being done when pressing the confirm, but in the case where you don't have anything saved and just start normally, shouldn't everything be init ?

Reply:
Hmm yeah good point, I originally thought that we would hit this block already initialized, but it doesn't hurt to include the init here as well.